src/wizard/wizard-dialog.ts

```diff
@@ -613,21 +595,35 @@ export class ESPHomeWizardDialog extends LitElement {
       this._state =
         supportsWebSerial && allowsWebSerial
           ? "connect_webserial"
-          : "pick_board";
+          : "pick_platform";
     }, 0);
   }
 
-  private _handlePickBoardRadio(ev: MouseEvent) {
-    this._board = (ev.target as any).value;
+  private _handleUseRecommendedCheckbox(ev: Event) {
+    this._useRecommended = (ev.target as HTMLInputElement).checked;
   }
 
-  private _handlePickBoardCustom(ev: Event) {
-    this._customBoard = (ev.target as HTMLSelectElement).value;
+  private _handlePickBoardSelect(ev: Event) {
+    this._board = (ev.target as HTMLSelectElement).value;
   }
 
-  private async _handlePickBoardSubmit() {
-    this._data.board =
-      this._board === "CUSTOM" ? this._customBoard : this._board;
+  private async _handlePickPlatformClick(ev: Event) {
+    this._platform = (ev.currentTarget as any).platform;
+    this._defaultBoard = DEFAULT_BOARD[this._platform];
```

Comment:
let's not store this in an instance variable. It's static and can always be looked up using `DEFAULT_BOARD[this._platform]`